src/lib.rs

```diff
@@ -608,138 +808,328 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> HotShot<TYPES::ConsensusType
-                }
+        async_spawn(
+            tasks::network_lookup_task(self.clone(), shut_down.clone())
+                .instrument(info_span!("HotShot network lookup task",)),
+        );
+
+        let (handle_channels, task_channels) = match self.inner.config.execution_type {
+            ExecutionType::Continuous => (None, None),
+            ExecutionType::Incremental => {
+                let (send_consensus_start, recv_consensus_start) = unbounded();
+                (Some(send_consensus_start), Some(recv_consensus_start))
             }
-        }
+        };
+
+        let consensus_task_handle = async_spawn(
+            tasks::view_runner(
+                self.clone(),
+                started.clone(),
+                shut_down.clone(),
+                task_channels,
+            )
+            .instrument(info_span!("Consensus task handle",)),
+        );
+
+        let (broadcast_sender, broadcast_receiver) = channel();
+
+        let handle = HotShotHandle {
+            sender_handle: Arc::new(broadcast_sender.clone()),
+            hotshot: self.clone(),
+            stream_output: broadcast_receiver,
+            storage: self.inner.storage.clone(),
+            shut_down,
+        };
+        *self.inner.event_sender.write().await = Some(broadcast_sender);
+
+        let mut background_task_handle = self.inner.background_task_handle.inner.write().await;
+        *background_task_handle = Some(TaskHandleInner {
+            network_broadcast_task_handle,
+            network_direct_task_handle,
+            committee_network_broadcast_task_handle: Some(committee_network_broadcast_task_handle),
+            committee_network_direct_task_handle: Some(committee_network_direct_task_handle),
+            consensus_task_handle,
+            shutdown_timeout: Duration::from_millis(self.inner.config.next_view_timeout),
+            run_view_channels: handle_channels,
+            started,
+        });
+
+        handle
     }
 
-    /// Handle an incoming [`DataMessage`] that directed at this node
-    #[allow(clippy::unused_async)] // async for API compatibility reasons
-    async fn handle_direct_data_message(
+    #[instrument(
+        skip(self),
+        name = "Handle broadcast consensus message",
+        level = "error"
+    )]
+    async fn handle_broadcast_consensus_message(
         &self,
-        msg: DataMessage<TYPES>,
-        _sender: TYPES::SignatureKey,
-    ) {
-        debug!(?msg, "Incoming direct data message");
-        match msg {
-            DataMessage::SubmitTransaction(_, _) => {
-                // Log exceptional situation and proceed
-                warn!(?msg, "Broadcast message received over direct channel");
-            }
-        }
-    }
+        msg: SequencingMessage<TYPES, I>,
+        sender: TYPES::SignatureKey,
+    ) {
+        // TODO validate incoming data message based on sender signature key
+        // <github.com/ExpressoSystems/HotShot/issues/418>
```

Comment:
The link is broken (`EspressoSystems`). This got closed, I think we're doing this? Maybe?